core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -142,7 +144,83 @@ object ReassignPartitionsCommand extends Logging {
         TopicAndPartition(topic, partition) -> replicas
       }
     }
-    (partitionsToBeReassigned, currentAssignment)
+    (partitionsToBeReassigned, currentAssignment, disableRackAware match {
```

Comment:
So you are suggesting to do the broker balance check in a separate class, and replace this line with something like?
```
val brokerBalanceCheck = new BrokerBalanceCheck(partitionsToBeReassigned, currentAssignment, disableRackAware)
(partitionsToBeReassigned, currentAssignment, brokerBalanceCheck.report())
```

Just wanted to make sure I fully understood your comment.